shared/test/fake_redis_client.rb

```diff
@@ -29,8 +29,15 @@ def del(key)
   end
 
   def hdel(key, field)
-    value = get_hash_for_key(key).delete(field)
-    value ? 1 : 0
+    if field.is_a?(Array)
+      field.reduce(0) do |memo, next_field|
```

Comment:
Updating to match your suggestion.

FYI, code examples render much better when "fenced" with triple-backticks ``` on the lines before and after:

```ruby
# Attempt to delete each request field and return the number actually deleted.
fields.reduce(0) do |num_deleted, field|
    num_deleted += 1 if hash.delete(field)
  end 
end
```